Symfony/CS/Tests/IntegrationTest.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+use Symfony\CS\Fixer;
+use Symfony\CS\FixerInterface;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * Test that parses and runs the fixture *.test files found in
+ * /Fixtures/Integration.
+ *
+ * @author SpacePossum
+ */
+class IntegrationTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var Fixer
+     */
+    private $fixer;
+
+    /**
+     * @dataProvider getTests
+     */
+    public function testIntegration($fileName, $testTitle, $fixers, $input, $expected = null)
+    {
+        $this->fixer = new Fixer(); // cannot be done in setUp(): https://github.com/sebastianbergmann/phpunit/issues/836
+        $dummyFile = new \SplFileInfo(__FILE__);
+        Tokens::clearCache();
+        $tokens = Tokens::fromCode($input);
+
+        $appliedFixers = $this->fixer->fixFileTokens($dummyFile, $tokens, $fixers);
+        if (null === $expected) {
+            if (0 === count($appliedFixers)) {
+                return;
+            }
+            $this->assertSame($input, $tokens->generateCode(), sprintf('No changes expected for "%s" in "%s", fixers applied "%s".', $testTitle, $fileName, implode(', ', $appliedFixers)));
+            // even if the code is the same, the fixers should not have been used
+            $this->fail(sprintf('No changes expected for "%s" in "%s", fixers applied "%s".', $testTitle, $fileName, implode(', ', $appliedFixers)));
```

Comment:
well, the point here is that if you have ``null === $expected`` the test *always* fail here